homeassistant/components/jellyfin/__init__.py

```diff
@@ -15,7 +15,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Jellyfin from a config entry."""
     hass.data.setdefault(DOMAIN, {})
 
-    client = create_client()
+    if entry.unique_id is None:
+        _LOGGER.error("Config entry was not created correctly")
+        return False
+
+    client = create_client(entry.unique_id)
```

Comment:
For users upgrading, `entry.unique_id` is the username, not the uuid which is created for new users.
Is that a problem?

Reply:
It's not really a problem at this point in time, but in the future it could become a problem when allowing multiple server instances to be connected. I've added the function to migrate the config entry to make sure it will always have the proper id configured.